Use native filepicker dialog instead of input file

danilopolani opened the pull request:
With this PR when choosing a private key for SSH tunneling a native dialog will be used instead of the `input file` one.

**Added**:

- Native dialog filepicker for SSH private key
- Input group CSS
- When choosing a private key, it will open by default `.ssh` folder (if it does not exists, it will fallback on the home folder, at least on unix)

**Fixes**:

- Bug on my PC (Ubuntu 17.04) returning error "Access denied" when navigating to `.ssh` folder
- Probably #32 too

Read more: https://www.electronjs.org/docs/api/dialog

rathboma commented:
@danilopolani are you using the `snap` by any chance?

rathboma commented:
I still like the idea of this change separate from the permissions issue. I'll do some testing and figure out how to encapsulate this in a component.

danilopolani commented:
Yeah, I'm using the snap version. I thought about creating a component from this, I can work on it tomorrow if you want (I guess it should applied to Sqlite DB selection too)

rathboma commented:
So your change won't fix anything for the snap version -- snaps are only allowed (by default) access to NON hidden folders in the home directory.

I'm curious if your change will fix the garbled fonts in Manjaro though.

I'll need to figure out the limitations of the Mac client separately. Like Snaps I also had to define some permissions up front, so I don't think simply switching to the electron-level picker will do it.

Please do componentize the picker, I think using the electron-picker is much better as the web file picker has some broad limitations that the electron one does not have.

danilopolani commented:
Ehy @rathboma , I updated the code creating an isolated component `FilePicker` to be re-used.

A few notes:

1. I added as props the common settings we need right now (`showHiddenFiles` and `defaultPath`), but we can extend/override other values as well with the `config` object prop.
2. Currently it does not support multiple files selection, because it would need some change in the UI too, but I think we will never need multiple files in the project.

I replaced the SQLite input file too with this, let me know if it's fine or if you need some more changes!

rathboma commented:
Awesome! Thanks so much. I'll review in the next few days
